### PR TITLE
SelectorBar Sample Page

### PR DESCRIPTION
**Pull request by karkarl**
## Description

Adding SelectorBar sample!

## Screenshots (if appropriate):
image

### CODE REVIEW

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 1]
```diff
@@ -1903,8 +1902,8 @@
           "Title": "Pivot",
           "ApiNamespace": "Microsoft.UI.Xaml.Controls",
           "Subtitle": "Presents information from different sources in a tabbed view.",
-          "ImagePath": "ms-appx:///Assets/ControlImages/Pivot.png",
-          "Description": "A Pivot allows you to show a collection of items from different sources in a tabbed view.",
+          "ImagePath": "ms-appx:///Assets/ControlImages/Placeholder.png",
+          "Description": "<p>Pivot is deprecated. Please use to <b>SelectorBar</b> and <b>SelectorBarItem.</b></p><p>A Pivot allows you to show a collection of items from different sources in a tabbed view.<p>",
```
Comment (RBrid) [inline WinUIGallery/DataModel/ControlInfoData.json:1906]: use to ==> use the

File: WinUIGallery/DataModel/ControlInfoData.json [hunk 2]
```diff
@@ -1918,9 +1917,33 @@
...
-            "ListView",
-            "GridView"
+            "SelectorBar"
+          ]
+        },
+        {
+          "UniqueId": "SelectorBar",
+          "Title": "SelectorBar",
+          "ApiNamespace": "Microsoft.UI.Xaml.Controls",
+          "Subtitle": "Presents information from different sources in a tabbed view.",
```
Comment (RBrid) [inline WinUIGallery/DataModel/ControlInfoData.json:1927]: "Presents information from different sources in a tabbed view."
"SelectorBar is used to modify the content shown by allowing users to select and switch between different sets of data."

These two points are a bit misleading, I think.  The SelectorBar just lets you pick 1 option out of N, in my opinion.

"The SelectorBar is used to present a set of options. The user can pick one of them."   Something like this more general description?

Comment (karkarl) [reply WinUIGallery/DataModel/ControlInfoData.json:1927]: "The SelectorBar is used to present a set of options. The user can pick one of them." Sounds like a RadioButton

Maybe something like this:
Subtitle: "Presents information from a small set of different sources. The user can pick one of them."
Description: "SelectorBar is used to modify the content shown by allowing users to select and switch between a small, finite set of data."

File: WinUIGallery/ControlPages/SelectorBarPage.xaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                    pageType = typeof(SamplePage4);
+                    break;
+                default:
+                    pageType = typeof(SamplePage5);
+                    break;
+            }
+
+            var slideNavigationTransitionEffect = currentSelectedIndex - previousSelectedIndex > 0 ? SlideNavigationTransitionEffect.FromRight : SlideNavigationTransitionEffect.FromLeft;
+
+            ContentFrame.Navigate(pageType, null, new SlideNavigationTransitionInfo() { Effect = slideNavigationTransitionEffect });
```
Comment (dotMorten) [inline WinUIGallery/ControlPages/SelectorBarPage.xaml:83]: Is this a good pattern to show? If you click back and forth between two pages over and over, you'll be creating a HUGE backstack of the same two pages over and over again.

File: WinUIGallery/ControlPages/SelectorBarPage.xaml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    mc:Ignorable="d">
+
+    <Page.Resources>
+        <DataTemplate x:Key="ColorsTemplate" x:DataType="media:SolidColorBrush">
+            <ItemContainer Background="{x:Bind}" Width="112" Height="82" Margin="4" />
+        </DataTemplate>
+    </Page.Resources>
+
+    <StackPanel>
+        <local:ControlExample x:Name="Example1" HeaderText="A Basic SelectorBar" Margin="0,0,24,0">
```
Comment (niels9001) [inline WinUIGallery/ControlPages/SelectorBarPage.xaml:19]:
```suggestion
        <local:ControlExample x:Name="Example1" HeaderText="A Basic SelectorBar">
```

File: WinUIGallery/ControlPages/SelectorBarPage.xaml [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+&lt;SelectorBar x:Name="SelectorBar1"&gt;
+    &lt;SelectorBarItem x:Name="SelectorBarItemRecent" Text="Recent" Icon="Clock" /&gt;
+    &lt;SelectorBarItem x:Name="SelectorBarItemShared" Text="Shared" Icon="Share" /&gt;
+    &lt;SelectorBarItem x:Name="SelectorBarItemFavorites" Text="Favorites" Icon="Favorite" /&gt;
+&lt;/SelectorBar&gt;
+                </x:String>
+            </local:ControlExample.Xaml>
+        </local:ControlExample>
+
+        <local:ControlExample x:Name="Example2" HeaderText="SelectorBar with Frame Slide Transitions" Margin="0,0,24,0">
```
Comment (niels9001) [inline WinUIGallery/ControlPages/SelectorBarPage.xaml:36]:
```suggestion
        <local:ControlExample x:Name="Example2" HeaderText="SelectorBar with Frame Slide Transitions">
```

File: WinUIGallery/ControlPages/SelectorBarPage.xaml [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+            ContentFrame.Navigate(pageType, null, new SlideNavigationTransitionInfo() { Effect = slideNavigationTransitionEffect });
+
+            previousSelectedIndex = currentSelectedIndex;
+        }
+                </x:String>
+            </local:ControlExample.CSharp>
+        </local:ControlExample>
+
+        <local:ControlExample x:Name="Example3" HeaderText="SelectorBar Displaying Different Collections Using ItemsView" Margin="0,0,24,0">
```
Comment (niels9001) [inline WinUIGallery/ControlPages/SelectorBarPage.xaml:100]:
```suggestion
        <local:ControlExample x:Name="Example3" HeaderText="SelectorBar Displaying Different Collections Using ItemsView">
```

**Review by niels9001 — APPROVED** [review]
No need to set manual margins here as it breaks vertical spacing between `ControlSamples`.

ComboBox page:
image

SelectorBar page:
image

File: WinUIGallery/ControlPages/SelectorBarPage.xaml (hunk 3 above)
Comment (karkarl) [reply WinUIGallery/ControlPages/SelectorBarPage.xaml:92]: Good callout. Added `IsNavigationStackEnabled="False"` to address this issue.

**Pull request merged**